src/codec_aom.c

```diff
@@ -744,20 +749,12 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
             }
         }
 
-        if (lossless) {
-            aom_codec_control(&codec->internal->encoder, AV1E_SET_LOSSLESS, 1);
-        }
-        if (encoder->maxThreads > 1) {
-            aom_codec_control(&codec->internal->encoder, AV1E_SET_ROW_MT, 1);
-        }
-        if (encoder->tileRowsLog2 != 0) {
-            int tileRowsLog2 = AVIF_CLAMP(encoder->tileRowsLog2, 0, 6);
-            aom_codec_control(&codec->internal->encoder, AV1E_SET_TILE_ROWS, tileRowsLog2);
-        }
-        if (encoder->tileColsLog2 != 0) {
-            int tileColsLog2 = AVIF_CLAMP(encoder->tileColsLog2, 0, 6);
-            aom_codec_control(&codec->internal->encoder, AV1E_SET_TILE_COLUMNS, tileColsLog2);
-        }
+        aom_codec_control(&codec->internal->encoder, AV1E_SET_LOSSLESS, lossless);
+        aom_codec_control(&codec->internal->encoder, AV1E_SET_ROW_MT, encoder->maxThreads > 1);
+        int tileRowsLog2 = AVIF_CLAMP(encoder->tileRowsLog2, 0, 6);
+        aom_codec_control(&codec->internal->encoder, AV1E_SET_TILE_ROWS, tileRowsLog2);
+        int tileColsLog2 = AVIF_CLAMP(encoder->tileColsLog2, 0, 6);
+        aom_codec_control(&codec->internal->encoder, AV1E_SET_TILE_COLUMNS, tileColsLog2);
```

Comment:
We should make these `aom_codec_control()` calls only when necessary. Is it possible to determine which calls are necessary when `updateConfig` is true?

The reason is that `aom_codec_control()` is not as cheap as it looks, if you trace its code. It will cause libaom to validate all of its configuration settings. See https://crbug.com/aomedia/2732.

Reply:
These options can also be set using `aom_codec_set_option()`.
If we only sync when these value changes, we can compare with previous value to find out which value needs to be updated; If these values should always be reflected in encoder, then we have to update them each time, as we can't know the real current value.